File: 04-ORIENTACAO_A_OBJETOS/JavaScript - NodeJS/Exercicio 1/Fazendo com função construtora/CaixaDeNumero.js
//Utilizando Função Construtora

let CaixaDeNumero = function(numeroSecreto){
    this.numeroSecreto = numeroSecreto

    this.ehPar = function(){
        return this.numeroSecreto%2 == 0
    }

    this.dizerTabuada = function(){
        let vetor = []
        for(let i=0; i<10; i++)
            vetor[i] = this.numeroSecreto * (i+1)
        return vetor
    }

    this.dizerQualNumero = function(){
        return this.numeroSecreto
    }
}

module.exports = CaixaDeNumero

File: 04-ORIENTACAO_A_OBJETOS/Java/Exercicio 1/CaixaDeNumero.java
/*
PrincipalCaixaDeNumero.java
*/

public class CaixaDeNumero {
    int numeroSecreto;
    
    boolean ehPar(){
        return this.numeroSecreto%2 == 0;
    }
    
    int[] dizerTabuada(){
        int[] vetor = new int[10];
        for(int i=1; i<=10; i++)
            vetor[i-1] = this.numeroSecreto * i;
        
        return vetor;
    }
    
    String dizerQualNumero(){
        return "O número secreto é: "+this.numeroSecreto;
    }
}
